W-13792719-cli-examples-kt (#1085)
mulesoft/docs-runtime-fabric: modules/ROOT/pages/deploy-to-rtf-cli.adoc

include::anypoint-cli::partial$rtf-instances.adoc[tag=summary,leveloffset=+1]

include::anypoint-cli::partial$rtf-instances.adoc[tag=commands,leveloffset=+1]

== Deploy a Mule App Example

> The following example shows how to deploy a Mule application to Runtime Fabric using Anypoint CLI.

Reviewer: ```suggestion
The following example shows how to deploy a Mule appl to Runtime Fabric using Anypoint CLI.
```

=== Configure your Anypoint CLI Space

> To configure your CLI space, you must download and install `anypoint-cli-v4`:

Reviewer: ```suggestion
To configure your CLI space, download and install `anypoint-cli-v4`:
```

> . Open CLI and run this command:

Reviewer: ```suggestion
. Open CLI, and run this command:
```
You can also omit "this command" if you'd like. The same comment applies to the below sections.

+
[source,copy]
----
npm install -g anypoint-cli-v4
----
+
[start=2]

> . Install runtime-mgr plugin by running this command:

Reviewer: ```suggestion
. Install the `runtime-mgr` plugin by running this command:
```

+
[source,copy]
----
anypoint-cli-v4 plugins:install anypoint-cli-runtime-mgr-plugin
----
+
[start=3]
. Set your credentials by running these commands:
+
[source,copy]
----
anypoint-cli-v4 conf username <username>
----
[source,copy]
----
anypoint-cli-v4 conf password <password>
----

> To set other parameters for your space, review the xref:anypoint-cli::auth.adoc[Authentication to the Anypoint Platform CLI] documentation.

Reviewer: ```suggestion
To set other parameters for your space, see xref:anypoint-cli::auth.adoc[Authentication to the Anypoint Platform CLI].
```

> After your CLI space is set, you can deploy and manage Mule applications to Runtime Fabric.

Reviewer: ```suggestion
After your CLI space is set, deploy and manage Mule applications to Runtime Fabric.
```

=== List Existing Apps

To list existing apps, run the following command:
[source,copy]
----
anypoint-cli-v4 runtime-mgr:application:list.
----

=== Deploy a Mule App

To deploy the Mule app, run the following command:

[source,copy]
----
runtime-mgr:application:deploy <appName> <targetId> <runtimeVersion> <artifactId>
----

> Using (test-mule-1) version as default version: v1.0.6
> deploying new-app01, the output log would be similar to this :

Reviewer: ```suggestion
Using `test-mule-1` version as default: v1.0.6 +
After deploying `new-app01`, the output log is:
```

╟─────────────────────┼──────────────────────────────────────────────╢
║ Target:             │ 734024b7-002b-40cd-92d2-b659175d0557         ║
╟─────────────────────┼──────────────────────────────────────────────╢
║ Artifact Id :       │ test-mule-1                                  ║
╟─────────────────────┼──────────────────────────────────────────────╢
║ Replicas:           │ 1                                            ║
╟─────────────────────┼──────────────────────────────────────────────╢
║ Update Strategy     │ rolling                                      ║
╟─────────────────────┼──────────────────────────────────────────────╢
║ Specs:              │ e45ab9b9-fe58-4387-9975-2ffcb91742df (DESIRED║
╟─────────────────────┼──────────────────────────────────────────────╢
║ Properties:         │                                              ║
╟─────────────────────┼──────────────────────────────────────────────╢
║ Secure Properties:  │                                              ║
╚═════════════════════╧══════════════════════════════════════════════╝
----

You can set the following extra parameters for you deployment:

> * Password and username/client id:  `--password <value> [--username <value> | --client_id <value> | ]`

Reviewer: ```suggestion
* Password and username/client ID:  `--password <value> [--username <value> | --client_id <value> | ]`
```

* Client secret:   `--client_secret <value>`
* Bearer: `--bearer <value>`
* Organization:  `--organization <value>`
* Environment: `--environment <value>`
* Host: `--host <value>`
* Collect metrics: `--collectMetrics`
* Asset version: `--assetVersion <value>`

> * Group id: `--groupId <value>`

Reviewer: ```suggestion
* Group ID: `--groupId <value>`
```

* Replicas: `--replicas <value>`
* Object store v2: `--objectStoreV2`
* Update strategy: `--updateStrategy <value>`
* Clustered: `--clustered`
* Last mile security: `--lastMileSecurity`
* Disable am log forwarding: `--disableAmLogForwarding`
* Property: `--property <key:value>`
* Secure property: `--secureProperty <key:value>`
* Properties file: `--propertiesFile <value>`
* Scope logging config file: `--scopeLoggingConfigFile <value>`

> * Forward ssl session: `--forwardSslSession`

Reviewer: ```suggestion
* Forward SSL session: `--forwardSslSession`
```

* Path rewrite: `--pathRewrite <value>`
* Public endpoints: `--publicEndpoints <value>`
* Instance type: `--instanceType <value>`

=== Edit Existing Mule App

> To modify an existing Mule application, you need the app ID. You can obtain the app ID from the list of applications or if you want to modify the app that you just deployed, the app ID is in the response of that deployment.

Reviewer: ```suggestion
To modify an existing Mule app, you need the app ID. Obtain the app ID from the list of applications. If you want to modify the app that you just deployed, the app ID is in the response of that deployment.
```

║ Runtime:            │ 4.4.0                                        ║
╟─────────────────────┼──────────────────────────────────────────────╢
║ Target:             │ 734024b7-002b-40cd-92d2-b659175d0557         ║
╟─────────────────────┼──────────────────────────────────────────────╢
║ Artifact Id :       │ test-mule-1                                  ║
╟─────────────────────┼──────────────────────────────────────────────╢
║ Replicas:           │ 1                                            ║
╟─────────────────────┼──────────────────────────────────────────────╢
║ Update Strategy     │ rolling                                      ║
╟─────────────────────┼──────────────────────────────────────────────╢
║ Specs:              │ e45ab9b9-fe58-4387-9975-2ffcb91742df (DESIRED║
╟─────────────────────┼──────────────────────────────────────────────╢
║ Properties:         │                                              ║
╟─────────────────────┼──────────────────────────────────────────────╢
║ Secure Properties:  │                                              ║
╚═════════════════════╧══════════════════════════════════════════════╝
----

> Additionally, you can add more parameters to deploy your Mule app. If you want your app to have 2 replicas instead of 1, you can write `--replicas 2`  at the end of your script.

Reviewer 2: ```suggestion
Additionally, you can add more parameters to deploy your Mule app. If you want your app to have two replicas instead of 1, you can write `--replicas 2`  at the end of your script.
```

* Group id: `--groupId <value>`
* Replicas: `--replicas <value>`
* Replica size: `--replicaSize <value>`
* Object store v2: `--objectStoreV2`
* Update strategy: `--updateStrategy <value>`
* Clustered: `--clustered`
* Last mile security: `--lastMileSecurity`
* Disable am log forwarding: `--disableAmLogForwarding`
* Property: `--property <key:value>`
* Secure property: `--secureProperty <key:value>`
* Properties file: `--propertiesFile <value>`
* Scope logging config file: `--scopeLoggingConfigFile <value>`
* Forward ssl session: `--forwardSslSession`
* Path rewrite: `--pathRewrite <value>`
* Public endpoints: `--publicEndpoints <value>`
* Instance type: `--instanceType <value>`

=== Edit Existing Mule App

To modify an existing Mule application, you need the app ID. You can obtain the app ID from the list of applications or if you want to modify the app that you just deployed, the app ID is in the response of that deployment.

You can change the same parameters that you set when deploying the Mule app.
For example, to add a secure property, run the following command: 

[source,copy]
----
anypoint-cli-v4 runtime-mgr:application:modify <appId>  --secureProperty <key:value> 
----

> === Stop, Start and Delete a Mule App

Reviewer 2: ```suggestion
=== Stop, Start, and Delete a Mule App
```
